pkg/kubelet/cm/util/cgroups.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+const (
+	minCPUShares = uint64(2)
+	maxCPUShares = uint64(262144)
+	minCPUWeight = uint64(1)
+	maxCPUWeight = uint64(10000)
+)
+
+type largeInteger interface {
+	~int64 | ~uint64
+}
+
+// Convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+func CPUSharesToCPUWeight[T largeInteger](cpuShares T) T {
```

Comment:
In existing code both `cpuShares` and `cpuWeight` are represented by `uint64` data type. So I am wondering what is the advantage of using generics here. Maybe I am missing something.

Reply:
The test code (`pod_resize.go` and `node_container_manager_test.go`) was converting types back-and-forth, so thought it could make the code a bit more readable.

I've now removed the generic bits and instead tried to slightly refactor these tests to avoid the type conversions.

Thanks for taking a look!